### ISSUE TITLE
Option to set visibility within breakpoints for a control

### ISSUE THREAD

**Issue by asemoller** [origo-map/origo#1208]
**Is your feature request related to a problem? Please describe.**
Sometimes you want a control to be available only when the screen is large and not when it is small or vice versa. There can be several reasons why you want it, such as functionality or that there can easily be too many buttons on small screens.

**Describe the solution you'd like**
It would be nice if you can set which breakpoints (s, m or l) a control should be visible (or not visible) within. It's basically the same problem that existed for embedded maps.

**Comment by jokd** [comment]
Another idea is to not just hide or show, but to be able place controls differently based on screen size. 
For instance when screen is small, put some stuff in the menu instead.
This relates to #1158

**Comment by MattiasSp** [comment]
Building on the simple config example from #1158, what would be easiest to read? A list of _acceptable_ screen/embedding/container placement combinations or, conversely, hiding the buttons for specific sizes and in embedded maps etc?

Example 1: Opt-in
```
{
  "name": "genericControl",
  "options": {
    "placement": [
      "screen": {
        "embedded": false,
        "screenSizes": ["m", "l"]
      },
      "menu": {
        "embedded": true,
        "screenSizes": ["s", "m", "l"]
      }
    ]
  }
},
```
Example 2: Opt-out
```
{
  "name": "genericControl",
  "options": {
    "placement": [
      "screen": {
        "hideWhenEmbedded": true,
        "hideOnScreenSizes": ["s"]
      },
      "menu"
    ]
  }
},
```

Both of these strategies make it tricky/impossible to make sure that the button is always shown on either the screen or in the menu, but never both. Not sure if this is a required use case.

**Comment by MattiasSp** [comment]
Perhaps control button placement options should be built into a sort of OrigoControl parent class which controls could inherit from? Then the handling of control button placement would be in a single place, and all controls could be passed button placement options on creation...

**Comment by asemoller** [comment]
No, we don't! But maybe @MattiasSp has a solution for this? (see his comments above) It's a useful feature though.
